### PR TITLE
TypeScript migration (partial).

### PR DESCRIPTION
**Pull request by doggan**
Working through some of the TypeScript migrations/hardening when running `yarn tsc --strict`.

`Found 3531 errors in 249 files. -> Found 3500 errors in 240 files.`

Related: https://github.com/actualbudget/actual/issues/1483

### CODE REVIEW

File: packages/loot-core/src/shared/transactions.ts [hunk 1]
```diff
@@ -164,7 +164,7 @@ export function addSplitTransaction(transactions, id) {
     if (!trans.is_parent) {
       return trans;
     }
```
Comment (doggan) [inline packages/loot-core/src/shared/transactions.ts:166]: Making `last` into a typed generic in `util.ts` bubbled up type issues to `transactions.ts`. This is the only location in the codebase using `last`, so it seemed most practical to just remove it.

File: packages/desktop-client/src/components/manager/ConfigServer.tsx [hunk 2]
```diff
@@ -77,15 +77,15 @@ export default function ConfigServer() {
   }
 
   async function onSkip() {
-    await setServerUrl(null);
+    await setServerUrl('');
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/manager/ConfigServer.tsx:80]: **🔨 warning:** ‏we need to keep `null` here.

https://github.com/actualbudget/actual/blob/a8a0f777e2369a9f74becabab9f1eb96d50cba16/packages/loot-core/src/server/main.ts#L1775

File: packages/desktop-client/src/components/manager/ConfigServer.tsx [hunk 3]
```diff
@@ -77,15 +77,15 @@ export default function ConfigServer() {
...
     await loggedIn();
     navigate('/');
   }
 
   async function onCreateTestFile() {
-    await setServerUrl(null);
+    await setServerUrl('');
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/manager/ConfigServer.tsx:86]: Same here

File: packages/desktop-client/src/components/sidebar/ItemContent.tsx [hunk 4]
```diff
@@ -21,7 +21,7 @@ function ItemContent({
   forceActive,
   children,
 }: ItemContentProps) {
-  return onClick ? (
+  return onClick || !to ? (
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/sidebar/ItemContent.tsx:24]: **💭 thought:** ‏I don't think we need this change.

File: packages/desktop-client/src/components/sidebar/ItemContent.tsx [hunk 5]
```diff
@@ -6,8 +6,8 @@ import View from '../common/View';
 
 type ItemContentProps = {
   style: CSSProperties;
-  to: string;
-  onClick: MouseEventHandler<HTMLDivElement>;
+  to?: string;
+  onClick?: MouseEventHandler<HTMLDivElement>;
```
Comment (MatissJanis) [inline packages/desktop-client/src/components/sidebar/ItemContent.tsx:10]: **❓ question:** ‏is it that there can be either `to` or `onClick`? But not both at the same time? If so - we could solve it by changing the types slightly.

(very poor naming chosen here intentionally as this is just an example)

```
type BaseProps = { style: CSSProperties };
type ItemContentProps = AProps | BProps;

type AProps = BaseProps & { to: string };
type BProps = BaseProps & { onClick: () => void }; 
```

**Review by MatissJanis — CHANGES_REQUESTED** [review]
Exciting to see all these TS improvements. I hope someday we will be fully TS compatible + strict :)

**Comment by doggan** [comment]
Thanks for the feedback. Going to close this PR for now since it is stale and some of the changes have already been addressed in other merged PRs.

**Pull request closed without merging**